lib/internal/modules/esm/loader.js

```diff
@@ -168,100 +213,254 @@ class Loader {
     const job = new ModuleJob(this, url, evalInstance, false, false);
     this.moduleMap.set(url, job);
     const { module } = await job.run();
+
     return {
       namespace: module.getNamespace(),
     };
   }
 
-  async import(specifier, parent) {
-    const job = await this.getModuleJob(specifier, parent);
-    const { module } = await job.run();
-    return module.getNamespace();
+  async getModuleJob(specifier, parentURL) {
+    const { format, url } = await this.resolve(specifier, parentURL);
+    let job = this.moduleMap.get(url);
+    // CommonJS will set functions for lazy job evaluation.
+    if (typeof job === 'function') this.moduleMap.set(url, job = job());
+
+    if (job !== undefined) return job;
+
+    const moduleProvider = async (url, isMain) => {
+      const { format: finalFormat, source } = await this.load(url, { format });
+
+      const translator = translators.get(finalFormat);
+
+      if (!translator) throw new ERR_UNKNOWN_MODULE_FORMAT(finalFormat);
+
+      return FunctionPrototypeCall(translator, this, url, source, isMain);
+    };
+
+    const inspectBrk = (
+      parentURL === undefined &&
+      getOptionValue('--inspect-brk')
+    );
+
+    job = new ModuleJob(
+      this,
+      url,
+      moduleProvider,
+      parentURL === undefined,
+      inspectBrk
+    );
+
+    this.moduleMap.set(url, job);
+
+    return job;
   }
 
-  hook(hooks) {
-    const {
-      resolve,
-      dynamicInstantiate,
-      getFormat,
-      getSource,
-      transformSource,
-      getGlobalPreloadCode,
-    } = hooks;
+  /**
+   *
+   * @param {string | string[]} specifiers Path(s) to the module
+   * @param {string} [parentURL] Path of the parent importing the module
+   * @returns {object | object[]} A list of module export(s)
+   */
+  async import(specifiers, parentURL) {
```

Comment:
What does this method do exactly? Should that be the first line of the JSDoc comment?

And `import` feels like a dangerous name for a method, even if it’s allowed.

Reply:
please do not change the name of this function as it maps to what using dynamic import against a loader would do.

Reply:
So! actually, this was very confusing to me at first, so I definitely hear you. I thought I did add a description, but apparently not (will do now).

This method is the entry into the loading world; it's not usually called directly, except to add loaders.

Reply:
Okay. To @bmeck’s point, maybe the comment should not only explain what it does but also why it’s named `import`, and why it shouldn’t be renamed 😄